datasets/elkarhizketak/elkarhizketak.py

```diff
@@ -0,0 +1,176 @@
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        data_dir = dl_manager.download_and_extract(_URLs)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": data_dir["train"],
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={
+                    "filepath": data_dir["validation"],
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={
+                    "filepath": data_dir["test"],
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath):
+        """Yields examples."""
+        with open(filepath, encoding="utf-8") as f:
+            elkarhizketak = json.load(f)
+            for section in elkarhizketak["data"]:
+                wiki_page_title = section.get("title", "").strip()
+                background = section.get("background", "").strip()
+                section_title = section.get("section_title", "").strip()
+
+                for dialogue in section["paragraphs"]:
+                    context = dialogue["context"].strip()
+                    dialogue_id = dialogue["id"]
+
+                    yesnos = []
+                    questions = []
+                    turn_ids = []
+                    answers = []
+                    orig_answers = {"texts": [], "answer_starts": []}
+
+                    for turn in dialogue["qas"]:
+                        yesnos.append(turn["yesno"])
+                        questions.append(turn["question"])
+                        turn_ids.append(turn["id"])
+
+                        ans_ = {
+                            "texts": [t["text"].strip() for t in turn["answers"]],
+                            "answer_starts": [t["answer_start"] for t in turn["answers"]],
+                            "input_texts": [t["input_text"] for t in turn["answers"]],
+                        }
+                        answers.append(ans_)
+
+                        orig_answers["texts"].append(turn["orig_answer"]["text"])
+                        orig_answers["answer_starts"].append(turn["orig_answer"]["answer_start"])
+
+                    yield dialogue_id, {
+                        "dialogue_id": dialogue_id,
+                        "wikipedia_page_title": wiki_page_title,
+                        "background": background,
+                        "section_title": section_title,
+                        "context": context,
+                        "turn_ids": turn_ids,
+                        "questions": questions,
+                        "yesnos": yesnos,
+                        "answers": answers,
+                        "orig_answers": orig_answers,
+                    }
```

Comment:
If I understand correctly, here you return one example = multiple questions

For consistency with other QA datasets like `squad`, could you change this part to return one example per question ?

e.g. `squad` has the following feature types:
```python
            features=datasets.Features(
                {
                    "id": datasets.Value("string"),
                    "title": datasets.Value("string"),
                    "context": datasets.Value("string"),
                    "question": datasets.Value("string"),
                    "answers": datasets.features.Sequence(
                        {
                            "text": datasets.Value("string"),
                            "answer_start": datasets.Value("int32"),
                        }
                    ),
                }
            ),
```

Reply:
Done!
This dataset is more similar to QuAC than SQuAD, and I followed the same structure (and feature types) as QuAC. But now I changed and it returns just one example

Reply:
Thanks ! We'll probably have to update QuAC at one point for consistency. Let me know if you'd be interested in updating QuAC's format. Using the same format allow users to simply change the dataset name and run the same evaluation code